### PR TITLE
feat: download snapshots from CloudFlare

### PR DESCRIPTION
**Pull request by lemmih**
## Summary of changes

Changes introduced in this pull request:

- Download snapshots from CloudFlare.
- Use `reqwest` instead of `aria2c`.
- Log to stderr in `forest-tool`.

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by lemmih** [comment]
`aria2c` is no longer needed now that the snapshots are downloaded from a CDN.

File: src/cli_shared/snapshot.rs [hunk 1]
```diff
@@ -69,10 +68,11 @@ pub async fn fetch(
     chain: &NetworkChain,
     vendor: TrustedVendor,
 ) -> anyhow::Result<PathBuf> {
-    let (_len, url) = peek(vendor, chain).await?;
-    let (date, height, forest_format) = ParsedFilename::parse_url(&url)
-        .context("unexpected url format")?
+    let (_len, path) = peek(vendor, chain).await?;
```
Comment (elmattic) [inline src/cli_shared/snapshot.rs:71]: Doc comment seems out of sync. It is stating that we try to download first with `aria2C`.

**Review by elmattic — APPROVED** [review]
Looks good to me.
Generally speaking, for user facing binaries(forest-cli/forest-tool) shouldn't we use progress bars and not progress logs?

File: src/cli_shared/snapshot.rs (hunk 1 above)
Comment (elmattic) [reply src/cli_shared/snapshot.rs:71]: Is `aria2c` still need to be a dependency of forest repository?

Comment (lemmih) [reply src/cli_shared/snapshot.rs:71]: `aria2c` is no longer used at all. I'll update the documentation.

**Review by hanabi1224 — APPROVED** [review]

**Comment by LesnyRumcajs** [comment]
> `aria2c` is no longer needed now that the snapshots are downloaded from a CDN.

Did you compare the speed of both approaches?

**Comment by lemmih** [comment]
> > `aria2c` is no longer needed now that the snapshots are downloaded from a CDN.
> 
> Did you compare the speed of both approaches?

My download speed in Europe from CloudFlare is top-notch. Hailong tested from China and got 12MiB/s from CloudFlare (using wget) and only 300KiB/s from DO (using `aria2c -x5`). CloudFlare has much better bandwidth than non-cdn DO.

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**